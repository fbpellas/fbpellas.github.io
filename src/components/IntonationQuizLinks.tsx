import * as React from 'react';
import { QuizIndex } from '../types';

export interface IntonationQuizLinksProps {
  setIndexCarousel: React.Dispatch<React.SetStateAction<QuizIndex>>;
  setPageAndClear: (hash: string) => void;
}

const IntonationQuizLinks: React.FC<IntonationQuizLinksProps> = (props) => {
  const { setIndexCarousel, setPageAndClear } = props;
  const { GuessThePattern, ShoppingForAPresent } = QuizIndex;

  return (
    <>
      <div>Test yourself to see how well you know intonations:</div>
      <ul>
        <li>
          <a
            className="clickable-page"
            href="#quiz"
            onClick={() => {
              setPageAndClear('quiz');
              setIndexCarousel(GuessThePattern);
            }}
          >
            Guess the Pattern
          </a>
        </li>
        <li>
          <a
            className="clickable-page"
            href="#quiz"
            onClick={() => {
              setPageAndClear('quiz');
              setIndexCarousel(ShoppingForAPresent);
            }}
          >
            Shopping for a Present
          </a>
        </li>
      </ul>
    </>
  );
};

export { IntonationQuizLinks };
