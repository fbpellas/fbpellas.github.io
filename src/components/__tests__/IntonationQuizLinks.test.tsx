import React from 'react';
import { render } from '@testing-library/react';
import { IntonationQuizLinks } from '../IntonationQuizLinks';

test('IntonationQuizLinks', () => {
  const fn = jest.fn();
  const { container } = render(<IntonationQuizLinks setIndexCarousel={fn} setPageAndClear={fn} />);
  expect(container).toMatchSnapshot();
});
