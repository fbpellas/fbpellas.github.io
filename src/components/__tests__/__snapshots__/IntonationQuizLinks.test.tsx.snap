// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntonationQuizLinks 1`] = `
<div>
  <div>
    Test yourself to see how well you know intonations:
  </div>
  <ul>
    <li>
      <a
        class="clickable-page"
        href="#quiz"
      >
        Guess the Pattern
      </a>
    </li>
    <li>
      <a
        class="clickable-page"
        href="#quiz"
      >
        Shopping for a Present
      </a>
    </li>
  </ul>
</div>
`;
